src/ILLink.RoslynAnalyzer/RequiresAssemblyFilesAnalyzer.cs

```diff
@@ -134,17 +142,53 @@ public override void Initialize (AnalysisContext context)
 					CheckCalledMember (operationContext, methodSymbol, dangerousPatterns);
 				}, OperationKind.DelegateCreation);
 
+				static void CheckStaticConstructors (OperationAnalysisContext operationContext,
+					ImmutableArray<IMethodSymbol> constructors)
+				{
+					foreach (var constructor in constructors) {
+						if (constructor.Parameters.Length == 0 && constructor.HasAttribute (RequiresAssemblyFilesAttribute) && constructor.MethodKind == MethodKind.StaticConstructor) {
```

Comment:
Why are we checking to see if the argument is a static constructor? Isn't that all this method takes?

Reply:
The way we call the function is with static constructors, but the method is open to receive MethodSymbol so I double-check that we have a constructors. Just in case someone calls it with methods, having said that, we own the analyzers so I don't know if it makes sense to assume that someone else will call this method in a different way since is private

Reply:
I don't think it's worth guarding private methods -- but in the case you need to, this is an example of a programming bug, not a recoverable error, so the only correct way to handle is to throw an uncaught exception (crash the process).

I'd recommend Joe Duffy's write-up here on the different between expected failures and logic errors: http://joeduffyblog.com/2016/02/07/the-error-model/#reliability-fault-tolerance-and-isolation